client/client-multi/client-cpp/impl/log/SQLiteDBLogStorage.cpp

```diff
@@ -101,13 +101,20 @@
     "SELECT MAX(" KAA_BUCKETS_SIZE_IN_RECORDS_FIELD_NAME "), MAX(" KAA_BUCKETS_SIZE_IN_BYTES_FIELD_NAME ")" \
     "FROM " KAA_BUCKETS_TABLE_NAME ";"
 
-#define KAA_GET_THE_OLDEST_BUCKET \
+#define KAA_GET_THE_OLDEST_UNUSED_BUCKET \
     "SELECT * FROM " KAA_BUCKETS_TABLE_NAME " " \
-    "WHERE " KAA_BUCKETS_STATE_FIELD_NAME " = 0 " \
+    "WHERE " KAA_BUCKETS_STATE_FIELD_NAME " = 0 AND " KAA_BUCKETS_SIZE_IN_RECORDS_FIELD_NAME " != 0 " \
     "ORDER BY " KAA_BUCKETS_INNER_BUCKET_ID_FIELD_NAME " ASC " \
     "LIMIT 1;"
 
-#define KAA_MARK_BUCKET_AS_IN_USE \
+#define KAA_GET_THE_LATEST_BUCKET \
+    "SELECT * FROM " KAA_BUCKETS_TABLE_NAME " " \
+    "WHERE " KAA_BUCKETS_OUTER_BUCKET_ID_FIELD_NAME " = " \
+                        "(SELECT MAX(" KAA_BUCKETS_OUTER_BUCKET_ID_FIELD_NAME ") " \
+                         "FROM " KAA_BUCKETS_TABLE_NAME ") " \
+    "LIMIT 1;"
```

Comment:
Seems that `LIMIT 1` is not needed, or you should use `ORDER BY` to get deterministic results